### PR TITLE
New mesh syntax in lua input files

### PR DESCRIPTION
**Pull request by ragusa**

### CODE REVIEW

File: resources/configure_dependencies.py [hunk 1]
```diff
@@ -465,14 +465,14 @@ def InstallPETSC(pkg: str, ver: str, gold_file: str):
             log_file.write(f"{command}\n{err}\n")
             package_log_file.write(f"{command}\n{err}\n")
 
-        command = f"make all -j{argv.jobs}"
+        command = f"make OMAKE_PRINTDIR=gmake all -j{argv.jobs}"
```
Comment (zachhardy) [inline resources/configure_dependencies.py]:
```suggestion
         make_command = "make"
         s, e, outstr = ExecSub(make_command + " --version")
         if outstr.find("GNU Make") > 0:
             make_command = "make OMAKE_PRINTDIR=gmake"
             
        command = f"{make_command} all -j{argv.jobs}"
```
Per @Naktakala  suggestion in Slack

File: resources/configure_dependencies.py [hunk 2]
```diff
@@ -465,14 +465,14 @@ def InstallPETSC(pkg: str, ver: str, gold_file: str):
...
         success, err = ExecSub(command, out_log=package_log_file, env_vars=env_vars)
         if not success:
             print(command, err)
             log_file.write(f"{command}\n{err}\n")
             package_log_file.write(f"{command}\n{err}\n")
 
-        command = "make install"
+        command = "make OMAKE_PRINTDIR=gmake install"
```
Comment (zachhardy) [inline resources/configure_dependencies.py]:
```suggestion
        command = "make install"
```
The fix is only needed for `make all`. If you had trouble without it here as well, feel free to disregard. I have never used this here and have never had trouble

File: test/modules/LinearBoltzmannSolvers/DSA/acceleration_diffusion_CFEM.lua [hunk 3]
```diff
@@ -1,33 +1,30 @@
...
-    zmesh[i] = zmin + k*dz
+    znodes[i] = zmin + k*dz
 end
 
---chiMeshCreateUnpartitioned3DOrthoMesh(mesh,mesh,mesh)
-chiMeshCreateUnpartitioned2DOrthoMesh(mesh,mesh)
---chiMeshCreateUnpartitioned1DOrthoMesh(mesh)
-chiVolumeMesherExecute();
+meshgen1 = chi_mesh.OrthogonalMeshGenerator.Create({ node_sets = {nodes,nodes} })
+-- meshgen1 = chi_mesh.OrthogonalMeshGenerator.Create({ node_sets = {nodes,nodes,znodes} })
```
Comment (zachhardy) [inline test/modules/LinearBoltzmannSolvers/DSA/acceleration_diffusion_CFEM.lua]: Remove, if unused

File: test/modules/LinearBoltzmannSolvers/DSA/acceleration_diffusion_CFEM.lua [hunk 4]
```diff
@@ -1,33 +1,30 @@
...
+znodes={}
 Nz=8
 Lz=1.6
 zmin=0.0
 dz=Lz/Nz
 for i=1,(Nz+1) do
     k=i-1
-    zmesh[i] = zmin + k*dz
+    znodes[i] = zmin + k*dz
 end
```
Comment (zachhardy) [inline test/modules/LinearBoltzmannSolvers/DSA/acceleration_diffusion_CFEM.lua]: Remove, if unused

File: test/modules/LinearBoltzmannSolvers/DSA/acceleration_diffusion_DFEM.lua [hunk 5]
```diff
@@ -1,33 +1,30 @@
...
+znodes={}
 Nz=8
 Lz=1.6
 zmin=0.0
 dz=Lz/Nz
 for i=1,(Nz+1) do
     k=i-1
-    zmesh[i] = zmin + k*dz
+    znodes[i] = zmin + k*dz
 end
```
Comment (zachhardy) [inline test/modules/LinearBoltzmannSolvers/DSA/acceleration_diffusion_DFEM.lua]: Remove, if unused

File: test/modules/LinearBoltzmannSolvers/DSA/acceleration_diffusion_DFEM.lua [hunk 6]
```diff
@@ -1,33 +1,30 @@
...
 
---chiMeshCreateUnpartitioned3DOrthoMesh(mesh,mesh,mesh)
-chiMeshCreateUnpartitioned2DOrthoMesh(mesh,mesh)
---chiMeshCreateUnpartitioned1DOrthoMesh(mesh)
-chiVolumeMesherExecute();
+meshgen1 = chi_mesh.OrthogonalMeshGenerator.Create({ node_sets = {nodes,nodes} })
+-- meshgen1 = chi_mesh.OrthogonalMeshGenerator.Create({ node_sets = {nodes,nodes,znodes} })
```
Comment (zachhardy) [inline test/modules/LinearBoltzmannSolvers/DSA/acceleration_diffusion_DFEM.lua]: Remove, if unused

**Review by zachhardy — APPROVED** [review]
This looks very tedious... Please see the requested changes in the configure dependencies script.

File: resources/configure_dependencies.py (hunk 2 above)
Comment (ragusa) [reply resources/configure_dependencies.py]: I followed your suggestion. Then I did a ./configure clean and ./configure. Then re-compile and passed all tests. If this is good enough, then we can go with your suggestion. I think I did a replace-all, that's why line 475 was changed as well.

Comment (Naktakala) [reply resources/configure_dependencies.py]: Just an FYI. The configure.sh only operates on the chi_build directory. Does nothing to dependencies. To really test it you should clean out your dependencies directory and use resources/configure_dependencies.py again.

File: resources/configure_dependencies.py [hunk 7]
```diff
@@ -465,7 +465,12 @@ def InstallPETSC(pkg: str, ver: str, gold_file: str):
             log_file.write(f"{command}\n{err}\n")
             package_log_file.write(f"{command}\n{err}\n")
 
-        command = f"make all -j{argv.jobs}"
+        make_command = "make"
+        s, e, outstr = ExecSub(make_command + " --version")
```
Comment (Naktakala) [inline resources/configure_dependencies.py]: This crashes. I know you based it on something I pasted in Slack but that copy paste was from code that had other changes.
Solution: Can we remove changes to this file from the PR? Included it in a different scope. I have the fix ready.

**Review by Naktakala — CHANGES_REQUESTED** [review]
This was VERY valuable. If we just remove the configure_dependencies.sh changes then this can move forward. I will fix up configure_dependencies.sh by this afternoon.

File: resources/configure_dependencies.py (hunk 7 above)
Comment (Naktakala) [reply resources/configure_dependencies.py]:
```
Traceback (most recent call last):
  File "/Users/vermjc/Repos/ChiTech/chi-tech/resources/configure_dependencies.py", line 600, in <module>
    success = InstallPETSC(pkg, ver, gold_file="include/petsc")
  File "/Users/vermjc/Repos/ChiTech/chi-tech/resources/configure_dependencies.py", line 469, in InstallPETSC
    s, e, outstr = ExecSub(make_command + " --version")
ValueError: not enough values to unpack (expected 3, got 2)
```

Comment (zachhardy) [reply resources/configure_dependencies.py]: Oops. Let's move it to its own PR.

**Review by Naktakala — APPROVED** [review]

**Pull request merged**